launch.go

```diff
@@ -3,237 +3,44 @@ package main // import "github.com/collisdigital/go-launch-a-survey"
-// TODO: Add support for password protected private keys
-func loadSigningKey() (interface{}, error) {
-	signingKeyPath := settings.GetSettings()["JWT_SIGNING_KEY_PATH"]
-	if keyData, err := ioutil.ReadFile(signingKeyPath); err == nil {
-		block, _ := pem.Decode(keyData)
-		priv, err := x509.ParsePKCS1PrivateKey(block.Bytes)
-		if err == nil {
-			return priv, nil
-		}
-	}
-	return nil, errors.New("Failed to load signing key")
-}
-
-func convertPostToToken(PostFields PostFields) (token string, err error) {
-	log.Println("POST received...", PostFields)
-
-	issued := time.Now()
-	expires := issued.Add(time.Minute * 10)
-
-	cl := EqClaims{
-		Claims: jwt.Claims{
-			IssuedAt: jwt.NewNumericDate(issued),
-			Expiry:   jwt.NewNumericDate(expires),
-			ID:       uuid.NewV4().String(),
-		},
-		UserId:                PostFields.UserId,
-		EqId:                  PostFields.EqId,
-		PeriodId:              PostFields.PeriodId,
-		PeriodStr:             PostFields.PeriodStr,
-		CollectionExerciseSid: PostFields.CollectionExerciseSid,
-		RuRef:          PostFields.RuRef,
-		RuName:         PostFields.RuName,
-		RefPStartDate:  PostFields.RefPStartDate,
-		RefPEndDate:    PostFields.RefPEndDate,
-		FormType:       PostFields.FormType,
-		ReturnBy:       PostFields.ReturnBy,
-		TradAs:         PostFields.TradAs,
-		EmploymentDate: PostFields.EmploymentDate,
-		RegionCode:     PostFields.RegionCode,
-		LanguageCode:   PostFields.LanguageCode,
-		VariantFlags:   PostFields.VariantFlags,
-		Roles:          PostFields.Roles,
-		TxId:           uuid.NewV4().String(),
-	}
-
-	signingKey, err := loadSigningKey()
-	if err != nil {
-		fmt.Printf("Error loading signing key; err: %v", err)
-		return
-	}
-
-	encryptionKey, err := loadEncryptionKey()
-	if err != nil {
-		fmt.Printf("Error loading encryption key; err: %v", err)
-		return
-	}
-
-	opts := jose.SignerOptions{}
-	opts.WithType("JWT")
-	opts.WithHeader("kid", "EDCRRM")
-
-	signer, err := jose.NewSigner(jose.SigningKey{Algorithm: jose.RS256, Key: signingKey}, &opts)
-	if err != nil {
-		fmt.Printf("Error creating JWT signer; err: %v", err)
-		return
-	}
-
-	encryptor, err := jose.NewEncrypter(
-		jose.A256GCM,
-		jose.Recipient{Algorithm: jose.RSA_OAEP, Key: encryptionKey},
-		(&jose.EncrypterOptions{}).WithType("JWT").WithContentType("JWT"))
+func serveTemplate(templateName string, data interface{}, w http.ResponseWriter, r *http.Request) {
+	lp := filepath.Join("templates", "layout.html")
+	fp := filepath.Join("templates", filepath.Clean(templateName))
 
-	if err != nil {
-		fmt.Printf("Error creating JWT encrypter; err: %v", err)
+	// Return a 404 if the template doesn't exist or is directory
+	info, err := os.Stat(fp)
+	if err != nil && (os.IsNotExist(err) || info.IsDir()) {
+		fmt.Println("Cannot find: " + fp)
+		http.NotFound(w, r)
 		return
 	}
 
-	token, err = jwt.SignedAndEncrypted(signer, encryptor).Claims(cl).CompactSerialize()
-
+	tmpl, err := template.ParseFiles(lp, fp)
```

Comment:
this is done on every request, could be cached

worth looking at [unrolled/render](https://github.com/unrolled/render), but it might be a bit of a heavy dependency for what you're doing